webofneeds/won-owner-webapp/src/main/webapp/app/actions/actions.js

```diff
@@ -354,6 +354,15 @@ export function needsClose(needUri) {
                 eventUri: data.eventUri,
                 message: data.message
             }));
+
+            //Close all the open connections of the need
+            const affectedConnections = getState().getIn(['needs', needUri, 'connections']).map(conn => conn && conn.get("uri"));
+            for (con in affectedConnections.toJSON()) {
+                if( getState().getIn(['needs', needUri, 'connections', con, 'state']) === won.WON.Connected) {
+                    dispatch(actionCreators.connections__close(con),
+                    )
+                }
+            }
```

Comment:
wouldnt it be better to dispatch the connections__close directly inside the .map function instead of getting all the affected connections making a mutable entity out of it and dispatching the action in a loop?

Reply:
If this is possible, yes we should do it like that. I did not think about it.